sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -766,6 +771,31 @@ object PushDownPredicate extends Rule[LogicalPlan] with PredicateHelper {
 
       project.copy(child = Filter(replaceAlias(condition, aliasMap), grandChild))
 
+    // Similar to the above Filter over Project
+    // LeftSemi/LeftAnti over Project
+    case join @ Join(project @ Project(projectList, grandChild), rightOp,
+                     LeftSemiOrAnti(joinType), joinCond)
```

Comment:
Is it optimized to push down the `Join` through the `Project`?

I think the `Project` helps prune the left side of the join.